sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -79,18 +79,6 @@ private[sql] class TypeCollection(private val types: Seq[AbstractDataType])
 private[sql] object TypeCollection {
 
   /**
-   * Types that can be ordered/compared. In the long run we should probably make this a trait
-   * that can be mixed into each data type, and perhaps create an `AbstractDataType`.
-   */
-  // TODO: Should we consolidate this with RowOrdering.isOrderable?
```

Comment:
Nope, `RowOrdering.isOrderable` (which is used by `TypeUtils.checkForOrderingExpr`) returns true on a strict superset of this type collection as it works for complex types that need recursive checks.